junit-common/src/main/java/org/jboss/weld/junit/AbstractWeldInitiator.java

```diff
@@ -136,6 +136,26 @@ protected ToInject createToInject(Object instanceToInject) {
         return new ToInject(instanceToInject);
     }
 
+    /**
+     * Injects the given non-contextual instance immediately. The returned {@link AutoCloseable} should be used
+     * to release the creational context once the injected beans are no longer needed.
+     *
+     * <p>Example:
+     * <pre>{@code
+     * try (AutoCloseable contextReleaser = inject(this)) {
+     *     // do some things with the injected instances
+     * }
+     * }</pre>
+     *
+     * @param target the target to inject
+     * @return an {@code AutoCloseable} to release the creational context
+     */
+    public AutoCloseable inject(Object target) {
```

Comment:
Do we really want to expose this method publicly? `AbstractWeldInitiator` is a basis for both, junit 4 and 5 initiators so this will be available anywhere. What's your take on this @mkouba and @Vampire?
Also, if we do keep it, then we need to have this tested and documented in both versions. And maybe renamed to something that indicates it is an injection into a non-contextual object?

Reply:
Well, my intention was for it to actually be usable to inject into any object and in both variants, that's why I put it to the base class and made it public. If you don't like that or don't like the name, I can change either of course.

Reply:
TBH I don't see a use case for this but that does not mean there isn't any ;-). That said, I'm +1 for this method iff we add a test and docs.

Reply:
I don't see an issue with the method staying (albeit I don't see a use case for it yet), I just wanted to make sure it was intentional. However, we should add some simple tests to junit4/5 just to have it covered.
As for name, `inject` is very broad. I'd prefer something that shows the method does injection into non-contextual objects - maybe `injectNonContextual(Object target)?

Reply:
I just took the existing method `org.jboss.weld.junit.AbstractWeldInitiator.AbstractBuilder#inject` as blueprint as it does the exact same thing.
I now changed the name of the new method as suggested and added two more simple tests that cover it.